From d9bb4b5cc1dba8dc6a78a1dcf47c758a1fec79cf Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Fri, 19 Jan 2018 14:44:09 +0800
Subject: [PATCH] drm/bridge/synopsys: Get mc_clkdis value from uboot.

When uboot logo is enabled, hdmi will not be setup again.
In this case, kernel driver will set wrong mc_clkdis value
when cec enable. So mc_clkdis should be got from uboot.

Change-Id: Idb9984d981489287966f2719f6ed4c729a183b58
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index d7cc7b7c634a..3616265dae0c 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -2627,6 +2627,7 @@ dw_hdmi_connector_atomic_flush(struct drm_connector *connector,
 		hdmi->hdmi_data.video_mode.mpixelclock = mode->clock;
 		hdmi->hdmi_data.video_mode.previous_pixelclock = mode->clock;
 		hdmi->hdmi_data.video_mode.previous_tmdsclock = mode->clock;
+		hdmi->mc_clkdis = hdmi_readb(hdmi, HDMI_MC_CLKDIS);
 		hdmi->phy.enabled = true;
 		return;
 	}
-- 
2.35.3

